ldp/nn/handlers/transformer_handler.py

```diff
@@ -0,0 +1,1017 @@
+            random_states = Path(ckpt).rglob("random_state*pkl")
+            for random_state in random_states:
+                random_state.unlink()
+
+    @staticmethod
+    def barrier() -> None:
+        if dist.is_initialized():
+            dist.barrier()
+
+
+class AsyncTransformer(TransformerHandler, AsyncTransformerInterface):
+    """Equivalent of AsyncTorchModule, but for a transformer model.
+
+    The model is instantiated in the current process.
+    """
+
+    def __init__(self, config: TransformerHandlerConfig):
+        # First init the local module
+        TransformerHandler.__init__(self, config)
+        # Then init the logic to buffer inputs to the module
+        AsyncTorchModule.__init__(
+            self,
+            module=self.module,
+            batch_size=config.batch_size,
+            max_wait_interval=config.max_wait_interval,
+            collate_fn=config.collate_fn,
+            decollate_fn=config.decollate_fn,
+            module_call_fn=config.module_call_fn,
+        )
+
+    async def __call__(
+        self,
+        inputs: str | BatchEncoding | list[dict] | None = None,
+        tools_json: list[dict] | None = None,
+        input_ids: torch.Tensor | None = None,
+        attention_mask: torch.Tensor | None = None,
+        **kwargs,
+    ) -> tuple[str, torch.Tensor]:
+        if inputs is None:
+            inputs = {"input_ids": input_ids, "attention_mask": attention_mask}
+        inputs_tokenized = _get_tokenized_inputs(self.tokenizer, inputs, tools_json).to(
+            device=_get_data_device()
+        )
+        inputs_len = inputs_tokenized["input_ids"].shape[1]
+        self.module.eval()
+        outputs = await AsyncTorchModule.__call__(self, **inputs_tokenized, **kwargs)
+        return _process_outputs(self.config, self.tokenizer, outputs, inputs_len)
+
+
+class ParallelWorkerConfig(FSDPConfig):
+    model_config = ConfigDict(extra="ignore")
+
+    rank: int = Field(description="Rank of the current process.")
+    world_size: int = Field(description="Total number of processes.")
+    local_rank: int = Field(description="Local rank of the current process.")
+    master_addr: str = Field(description="Address of the master node.")
+    master_port: int = Field(description="Port of the master node.")
+
+    def set_env_vars(self):
+        # These inform torch.distributed how to set up the process group
+        os.environ["CUDA_VISIBLE_DEVICES"] = str(self.local_rank)
+        os.environ["RANK"] = str(self.rank)
+        os.environ["WORLD_SIZE"] = str(self.world_size)
+        os.environ["LOCAL_RANK"] = str(self.local_rank)
+        os.environ["MASTER_ADDR"] = self.master_addr
+        os.environ["MASTER_PORT"] = str(self.master_port)
+
+        # These tell PretrainedModel.from_pretrained() that we're using FSDP
+        os.environ["ACCELERATE_USE_FSDP"] = "1"
+        os.environ["FSDP_CPU_RAM_EFFICIENT_LOADING"] = str(
+            int(self.cpu_ram_efficient_loading)
+        )
```

Comment:
Feel free to ignore this, but for huge env var updates, I like bulk updates `os.environ.update({...})` as opposed to individual sets